### PR TITLE
EVG-3646 test regression notifications by name

### PR DESCRIPTION
**Pull request by john-m-liu**
The UI is inconsistent with the other regexes, but I feel like it's fine

### CODE REVIEW

File: trigger/task.go [hunk 1]
```diff
@@ -558,6 +559,22 @@ func isTaskStatusRegression(oldStatus, newStatus string) bool {
 	return false
 }
 
+func (t *taskTriggers) testMatchesRegex(testName string, sub event.Subscription) bool {
```
Comment (richardsamuels) [inline trigger/task.go]: Nit, this doesn't need to be a method on taskTriggers, and you shouldn't be copying the whole subscription structure.

File: public/static/js/projects.js [hunk 2]
```diff
@@ -86,6 +86,9 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location,
       resource_type: "TASK",
       label: "a previously passing test in a task fails",
       regex_selectors: taskRegexSelectors(),
+      extraFields: [
+        {text: "Test names matching regex", key: "test-regex", validator: null}
```
Comment (richardsamuels) [inline public/static/js/projects.js]: Are empty regexes rejected?

File: model/event/subscriptions.go [hunk 3]
```diff
@@ -309,6 +313,14 @@ func validatePositiveFloat(s string) error {
 	return nil
 }
 
+func validateRegex(s string) error {
+	regex, err := regexp.Compile(s)
+	if regex == nil || err != nil {
+		return errors.New("invalid regex")
```
Comment (richardsamuels) [inline model/event/subscriptions.go]: errors.Wrap the err?

Comment (john-m-liu) [reply model/event/subscriptions.go]: fixed

File: public/static/js/projects.js (hunk 2 above)
Comment (john-m-liu) [reply public/static/js/projects.js]: No, but I'm doing this validation only on the server

File: trigger/task.go (hunk 1 above)
Comment (john-m-liu) [reply trigger/task.go]: fixed

**Pull request merged**